Marksman/Refs.fs

```diff
@@ -316,6 +334,7 @@ module Dest =
                 | Dest.LinkDef _ -> [ srcDoc ]
                 | Dest.Heading _
                 | Dest.Doc _ -> Folder.docs folder |> List.ofSeq
+                | Dest.Tag _ -> [ srcDoc ]
 
             let referencingEls =
```

Comment:
This is the list of references. To fit your functionality here you may want to match on the `declToFind` on line 342, e.g.
```fsharp
for targetDoc in targetDocs do
    match declToFind with
    | Dest.Tag(_, tag) -> 
        Doc.index targetDoc
        |> Index.filterTagsByName tag.name.text
        |> Seq.map (fun t -> Dest.Tag(targetDoc, t))
    | _ ->
        // existing code for links
```

Reply:
I had to adapt your example here as the required return type for `referencingEls` is `seq<Doc * Element * []<Dest>>` and not `seq<Dest>`. The full code would look like this:

```fsharp
let referencingEls =
    seq {
        for targetDoc in targetDocs do
            match declToFind with
            | Dest.Tag(_, tag) ->
                let targets =
                    Doc.index targetDoc
                    |> Index.filterTagsByName tag.text
                    |> Seq.map (fun t -> Dest.Tag(targetDoc, t))
                    |> Seq.toArray
                targetDoc, T(tag), targets
            | _ ->
                let targetDocRefs = resolveLinks folder targetDoc

                let backRefs =
                    findReferencingElements targetDocRefs declToFind
                    |> Seq.map (fun (el, dest) -> targetDoc, el, dest)

                yield! backRefs
    }
```